### PR TITLE
re-enable events behind `disable-collect-events` feature flag (for use in our own test suites)

### PR DESCRIPTION
**Pull request by Domiii**
* https://linear.app/replay/issue/RUN-1251/add-an-eventsenabled-feature-flag-to-chromium

After some first tests, I can see that:

1. Multiple events can be in-flight at the same time (which might be problematic for other reasons).
2. None are hitting `recordreplay::AreEventsDisallowed()` thus far.

### CODE REVIEW

File: third_party/blink/renderer/core/inspector/inspector_dom_debugger_agent.cc [hunk 1]
```diff
@@ -962,24 +962,44 @@ static String MakeReplayEventType(const String& eventTypeRaw,
...
-  // Disabled, see https://linear.app/replay/issue/RUN-1206
-  //recordreplay::OnEvent(qualifiedEventName.Ascii().c_str(), true);
-  (void)qualifiedEventName;
+  // Disabled by default, see https://linear.app/replay/issue/RUN-1251
+  if (recordreplay::IsRecordingOrReplaying() &&
+      !recordreplay::FeatureEnabled("disable-collect-events")) {
+    recordreplay::Print("[RUN-1251] DEBUG-EVENTS event START %d %d %s",
+                        recordreplay::AreEventsDisallowed(),
+                        gIsEventInFlight,
+                        replayEventType.Ascii().c_str());
```
Comment (bhackett1024) [inline third_party/blink/renderer/core/inspector/inspector_dom_debugger_agent.cc:978]: Debugging print statements like this should be removed.

File: third_party/blink/renderer/core/inspector/inspector_dom_debugger_agent.cc [hunk 2]
```diff
@@ -962,24 +962,44 @@ static String MakeReplayEventType(const String& eventTypeRaw,
...
-  // Disabled, see https://linear.app/replay/issue/RUN-1206
-  //recordreplay::OnEvent(qualifiedEventName.Ascii().c_str(), false);
-  (void)qualifiedEventName;
+  // Disabled by default, see https://linear.app/replay/issue/RUN-1251
+  if (recordreplay::IsRecordingOrReplaying() &&
+      !recordreplay::FeatureEnabled("disable-collect-events")) {
+    recordreplay::Print("[RUN-1251] DEBUG-EVENTS event END %d %d %s",
+                        recordreplay::AreEventsDisallowed(),
+                        gIsEventInFlight,
+                        replayEventType.Ascii().c_str());
```
Comment (bhackett1024) [inline third_party/blink/renderer/core/inspector/inspector_dom_debugger_agent.cc:997]: ditto

**Review by bhackett1024 — APPROVED** [review]

File: third_party/blink/renderer/core/inspector/inspector_dom_debugger_agent.cc [hunk 3]
```diff
@@ -962,24 +960,37 @@ static String MakeReplayEventType(const String& eventTypeRaw,
...
+                             bool isCallback) {
+  String replayEventType =
       MakeReplayEventType(eventName, eventTarget, isCallback);
-  // Disabled, see https://linear.app/replay/issue/RUN-1206
-  //recordreplay::OnEvent(qualifiedEventName.Ascii().c_str(), true);
-  (void)qualifiedEventName;
+
+  // Disabled by default, see https://linear.app/replay/issue/RUN-1251
+  if (recordreplay::IsRecordingOrReplaying() &&
+      !recordreplay::FeatureEnabled("disable-collect-events")) {
```
Comment (jasonLaster) [inline third_party/blink/renderer/core/inspector/inspector_dom_debugger_agent.cc:973]:
```
  if (recordreplay::IsRecordingOrReplaying() &&
      !recordreplay::FeatureEnabled("disable-collect-events")) {
```

Not a big deal, but it is typically easier to think in terms of feature flags enabling features as opposed to double negatives...

```
  if (recordreplay::IsRecordingOrReplaying() && recordreplay::FeatureEnabled("collect-events")) {
```

To your point about the feature being disabled by default. It's totally fine for `collect-events` to be `false` by default :) Feels good to turn things on 😉

Comment (Domiii) [reply third_party/blink/renderer/core/inspector/inspector_dom_debugger_agent.cc:973]: > it is typically easier to think in terms of feature flags enabling features as opposed to double negatives...

No one wants double negatives, but that design is not really up to me :cry:. It was done by @bhackett1024 and he does not want to "disabled defaults". Hence, we cannot actually change the defaults. I am following his lead on this. Several other features were added in the past with a `no-` and `disabled-` prefix for the same reason.

Regardless, Brian already agreed that we need a better and more general-purpose API [here](https://github.com/replayio/backend/pull/7030#pullrequestreview-1283354760), so I filed a follow up issue for this [here](https://linear.app/replay/issue/RUN-1263/feature-flag-api-for-runtimes).

Comment (Domiii) [reply third_party/blink/renderer/core/inspector/inspector_dom_debugger_agent.cc:973]: @jasonLaster [here](https://github.com/replayio/chromium-v8/blob/31707b5e1933d9ca33d3b95a9f82f610698c0da2/src/api/api.cc#L10959) is the list of available flags.

**Pull request merged**